crates/ln-dlc-node/src/node/invoice.rs

```diff
@@ -223,18 +224,32 @@ where
         Ok(route_hint_hop)
     }
 
-    pub fn send_payment(&self, invoice: &Invoice) -> Result<()> {
-        let (status, err) = match pay_invoice(invoice, Retry::Attempts(10), &self.channel_manager) {
+    pub fn pay_invoice(&self, invoice: &Invoice, amount: Option<u64>) -> Result<()> {
+        let (result, amt_msat) = match invoice.amount_milli_satoshis() {
```

Comment:
I'm not sure I follow this logic: 
Are you saying: if there is no amount in the `invoice`, then take the `amount`, else, take the `invoice`'s amount. 
However, you let the user enter an `amount` in the frontend nevertheless?

Reply:
Yes, so if the invoice does not have an amount we expect an amount to be provided. It's optional because I do not know upfront if this is a zero amount invoice or not.